karmagrambot/__init__.py

```diff
@@ -133,13 +133,37 @@ def opt_out(bot, update):
 
     message.reply_text('You are no longer being tracked in this chat ;)')
 
+def get_karma(bot, update):
+    message = update.message
+    chat_id = message.chat_id
+    user_id = message.from_user.id
+    text = message.text
+    username = message.from_user.username
+
+    if message.reply_to_message is not None:
+        user_id = message.reply_to_message.from_user.id
+        username = message.reply_to_message.from_user.username
+    
+    if len(text.split()) > 1:
+        username = text.split()[1]
+        db = dataset.connect(DB_URI)
+        users = db['users'].find(username=username)
+        for user in users:
+            user_id = user['user_id']
+        username = 'This user'
+
+    karma = analytics.get_karma(user_id, chat_id)
+
+    message.reply_text(f'@{username} has {karma} karma')
+
```

Comment:
I have a refactored version that I think adds more readability by splitting the responsibility between deciding how we want to decide whose karma is wanted and replying it (`get_karma`) and getting the user info for that (the other two functions).

```suggestion
def karma_id_with_message(message):
    target_message = message.reply_to_message if message.reply_to_message is not None else message

    user_id = target_message.from_user.id
    username = target_message.from_user.username

    return user_id, username


def karma_id_with_username(message, username):
    db = dataset.connect(DB_URI)
    try:
        [user,] = db['users'].find(username=username)
    except IndexError:
        return None

    user_id = user['user_id']

    return user_id, username


def get_karma(bot, update):
    message = update.message
    text = message.text
    username = message.from_user.username

    _, *args = text.split()

    user_id, username = karma_id_with_message(message) if args else karma_id_with_username(args[1])

    if user_id is None:
        message.reply_text(f"Could not find data about user {username}.")
        return

    karma = analytics.get_karma(user_id, message.chat_id)

    message.reply_text(f'@{username} has {karma} karma')
```